### ISSUE TITLE
Rails 4.2: simple_format omits sanitizing by default

### ISSUE THREAD

**Issue by john-999** [rails/rails#19944]
simple_format omits sanitizing by default (which should only happen when "sanitize: false" is used)

    <%= simple_format("<b>This text should not be bold</b>", {}, wrapper_tag: 'p') %>

**Comment by imanel** [comment]
It looks like default sanitization is very permissive:

1. [simple_format calls sanitize](https://github.com/rails/rails/blob/61f1b86b8493b66f0c3c80b0738630344764d6de/actionview/lib/action_view/helpers/text_helper.rb#L286)
2. [sanitize will call default white_list_sanitizer](https://github.com/rails/rails/blob/61f1b86b8493b66f0c3c80b0738630344764d6de/actionview/lib/action_view/helpers/sanitize_helper.rb#L84)
3. [that by default is `Rails::Html::Sanitizer`](https://github.com/rails/rails/blob/61f1b86b8493b66f0c3c80b0738630344764d6de/actionview/lib/action_view/helpers/sanitize_helper.rb#L126)
4. [that will call PermitScrubber](https://github.com/rails/rails-html-sanitizer/blob/7cbe2d30af5c81fa2282bd453f7aa62dd943473b/lib/rails/html/sanitizer.rb#L102)
5. [that will check if you have set tags options - if not it will call `Loofah::HTML5::Scrub.allowed_element?`](https://github.com/rails/rails-html-sanitizer/blob/7cbe2d30af5c81fa2282bd453f7aa62dd943473b/lib/rails/html/scrubbers.rb#L86-L92)
6. [that will check allowed table](https://github.com/flavorjones/loofah/blob/3753d0f578069ff05945838adbf713d9b7d080b7/lib/loofah/html5/scrub.rb#L14)
7. [which is insanely permissive](https://github.com/flavorjones/loofah/blob/3753d0f578069ff05945838adbf713d9b7d080b7/lib/loofah/html5/whitelist.rb)

So basically anything except of selected few (script, object, embed) will pass. Even SVG  and VIDEO is acceptable...

**Comment by maclover7** [comment]
I agree with @imanel -- the `<b>` tag is not removed from the string when being sanitized, as it is in the [allowed tags list](https://github.com/rails/rails-html-sanitizer/blob/v1.0.3/lib/rails/html/sanitizer.rb#L107). If you want to check out the reproduction script I've been using, it's available [here](https://gist.github.com/maclover7/042b37f5758019c5057ceec259d560c9). Closing since this isn't a bug.

**Comment by sfcgeorge** [comment]
This feels like a bug as it's unexpected, but the tests seem to show this is the correct behaviour.

* The current behaviour is by default `simple_format` allows most tags including weird ones like `<video>`, it does strip `<script>` though.   
* Setting `sanitize: false` allows everything.  
* There are no other settings. There is no way to say "just do the paragraph insertion on newlines" or "only allow `%w(strong em)`".  
* The documentation doesn't mention that any tags will get through, none of the examples show tags being allowed through, in fact one example shows the `<blink>` tag being stripped (one of the few that is) so to me the docs heavily imply that `simple_format` is supposed to replace `\n\n` with `<p>` and strip everything else.

There are examples in the tests of `<script>` being removed and `<b>` being allowed through, so I guess this is the intended behaviour. https://github.com/rails/rails/blob/fe1f4b2ad56f010a4e9b93d547d63a15953d9dc2/actionview/test/template/text_helper_test.rb#L45

It's surprising to me as the way I read the documentation I thought this method was for formatting say a user supplied description simply replacing newlines paragraphs. Is there another method for that? Or are we supposed to use `simple_format` with `sanitize` as below? I feel like a doc update to clarify what this method is intended for and how it should be used with regards to tags in user input would be really helpful at least. 

```ruby
<%= sanitize simple_format(@description), tags: %w(strong em), attributes: [] %>
```
